syntax = "proto3";
package permission;




import "google/protobuf/struct.proto";

/*
PutPermission请求
*/
message PutPermissionRequest {
    /*
    权限点id
    */
    string id = 1 ;
    /*
    系统名
    */
    string system = 2 ;
    /*
    需要鉴权的动作, 全局唯一
    */
    string action = 3 ;
    /*
    权限备注
    */
    string remark = 4 ;
    /*
    权限角色
    */
    repeated string roles = 5 ;
    /*
    单个资源结构参考PermissionResource. 如果依赖多个资源，资源间判定逻辑条件用%and或者%or来扩住。 比如： { "%or":[ { "system":"flow", "condition":{ "executeAuthorizers":"%user" }, "name":"flow" }, { "system":"tool", "condition":{ "executeAuthorizers":"%user" }, "name":"tool" } ] } 表示这个权限点依赖流程的执行授权白名单和工具的执行授权白名单才能鉴权成功。
    */
    google.protobuf.Value resource = 6 ;
}

/*
PutPermission返回
*/
message PutPermissionResponse {
    /*
    修改权限点数量
    */
    int32 count = 1 ;
}

/*
PutPermissionApi返回
*/
message PutPermissionResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    permission.PutPermissionResponse data = 4 ;
}

